framework/LearningGate.py

```diff
@@ -173,6 +217,393 @@ def train(self,trainingSet):
+      picker = slice(subdiv[0], subdiv[-1]+1)
+      ## TODO only consider one sample at 0? Should do more for non-ARMA ROMs!
+      ##  -> for now, only ARMAs can be used with this method, so we can address this later
+      data = dict((var,[trainingSet[var][0][picker]]) for var in trainingSet)
+      targetData = dict((var,data[var][0]) for var in targets)
+      self.targetDatas.append(targetData) # DEBUGG only
+      # create a new ROM and train it
+      newRom = copy.deepcopy(templateRom)
+      self.raiseADebug('Training segment',i,picker)
+      newRom.train(data)
+      roms.append(newRom)
+      # if clustering, evaluate metrics
+      if strategy in ['continuous','cluster']:
+        ## -> basic metrics (mean, variance, etc)
+        basicData = self._evaluateBasicMetrics(targetData)
+        for feature, val in basicData.items():
+          if i == 0:
+            clusterFeatureDict[feature] = [val]
+          else:
+            clusterFeatureDict[feature].append(val)
+        ## -> user-provided metrics
+        ### TODO self._evaluateMetrics()
+        ## -> ROM metrics
+        romData = newRom.getRomClusterValues(self._romClusterFeatureTemplate)
+        if i == 0:
+          for feature,val in romData.items():
+            clusterFeatureDict[feature] = [val]
+        else:
+          for feature,val in romData.items():
+            clusterFeatureDict[feature].append(val)
+    # fix rom list type
+    roms = np.array(roms)
+    return clusterFeatureDict, roms
+
+  def _subdivideDomain(self, clusterParams, clusterLengths, trainingSet, pivotParam):
+    """
+      Creates markers for subdividing the pivot parameter domain, either based on number of subdivisions
+      or on requested pivotValue lengths.
+      @ In, clusterParams, dict, dictionary of inputs/indices to cluster on mapped to number of subdivisions to make
+      @ In, clusterLengths, dict, dictionary of inputs/indices to cluster on mapped to length of pivot values to include
+      @ In, trainingSet, dict or list, data used to train the ROM; if a list is provided a temporal ROM is generated.
+      @ In, pivotParam, str, name of pivot parameter variable on which do subdivide
+      @ Out, counter, list(tuple), indices that belong to each division; at minimum (first index, last index)
+      @ Out, unclustered, list(tuple), as "counter" but for segments that will not be clustered
+    """
+    # subdivide domain
+    unclustered = [] # data ranges that didn't get clustered because they are particular
+    if len(clusterParams):
+      # segment by equal spacing
+      index, segments = clusterParams.items()[0]
+      dataLen = len(trainingSet[index][0])
+      self._romClusterInfo['historyLength'] = dataLen
+      ## TODO assumption: ARMA only trains on a single sample
+      counter = np.arange(dataLen)
+      counter = np.array_split(counter, segments)
+    else:
+      # segment by value
+      ## TODO assumption: ARMA only trains on a single sample
```

Comment:
I don't know, I don't have an example of doing this in another ROM right now. Otherwise, I would spend more time trying to make it generic without trying to fabricate a use case.